SynchronizationClock.md

```diff
@@ -49,18 +48,58 @@ ISR(TCD0_OVF_vect, ISR_NAKED)
             case 7:
                 USARTD1_DATA = *timestamp_byte2;
                 break;
+            // The final byte is dispatched much later than the previous 5.
             case 1998:
                 USARTD1_DATA = *timestamp_byte3;
                 break;
         }
     }
 ```
 
+Example of a microcontroller C++ code for converting the four received encoded bytes to the timestamp:
+````C
+    #define HARP_SYNC_OFFSET_US (672)
+
+    // Assume 4 bytes of timestamp data (without header) have been written to this array.
+    alignas(uint32_t) volatile uint8_t sync_data_[4];
+
+    // reinterpret 4-byte sequence as a little-endian uint32_t.
+    uint32_t encoded_sec = *(reinterpret_cast<uint32_t*>(self->sync_data_));
+    // Convert received timestamp to the current time in microseconds.
+    // Add 1[s] per protocol spec since 4-byte sequence encodes the **previous** second.
+    uint64_t curr_us = ((static_cast<uint64_t>(encoded_sec) + 1) * 1e6) - HARP_SYNC_OFFSET_US;
+````
+
+A full example demonstrating a state machine receiving the 6-byte sequence can be found in the [Pico Core](https://github.com/AllenNeuralDynamics/harp.core.rp2040/blob/main/firmware/src/harp_synchronizer.cpp).
+
+---
+
+
 ## Physical connection
 
-The physical connection is made by a simple audio cable. In the same folder of this file, you can find an [example](./PhysicalConnector.pdf) of the sender and the receiver.
+The physical connection is made by a simple 3.5mm audio cable.
+
+The connector pinout for a device *receiving* the timestamp is shown below:
+
+!["SynchReceiverSchematic](./assets/harp_clock_sync_receiver.png)
+
+A TVS diode is also suggested for ESD protection.
+
+> [!IMPORTANT]
+> The device receiving the timestamp must provide 3.3-5V (~10mA) on the audio jack **R** pin.
+
+The schematic snippet for a device *sending* the timestamp is shown below:
+
+!["SynchSenderSchematic](./assets/harp_clock_sync_sender.png)
```

Comment:
```suggestion
![SynchSenderSchematic](./assets/SynchSenderSchematic.png)
```

Could we rename the file to follow PascalCasing for consistency with other asset and document files? I like the name you used for the image alt tag so placing that as a suggestion.

By the way this did lead me into a rabbit hole to try and decide between "synch" and "sync" (https://grammarhow.com/sync-vs-synch/). I am happy to go with one or the other but we probably should be consistent.

Reply:
Done!